src/message_type.rs

```diff
@@ -17,7 +17,164 @@
 
 //! Message types for PeerMessages
 
+#![allow(unknown_lints, derive_hash_xor_eq)]
+
 use std::fmt;
+use std::hash::{Hash, Hasher};
+
+use hex;
+use sawtooth_sdk::consensus::engine::PeerMessage;
+
+use crypto::digest::Digest;
+use crypto::sha2::Sha512;
+use error::PbftError;
+use protos::pbft_message::{PbftBlock, PbftMessage, PbftMessageInfo, PbftViewChange};
+
+#[derive(Debug, Clone, PartialEq, Eq)]
+pub enum PbftMessageWrapper {
+    PbftMessage(PbftMessage),
+    PbftViewChange(PbftViewChange),
+}
+
+#[derive(Debug, PartialEq, Eq, Clone)]
+pub struct ParsedPeerMessage {
```

Comment:
Could you describe the purpose of this wrapper type? Is it for convenience or does it provide necessary functionality?

Reply:
It's necessary, as it holds information that we need to use to generate the consensus seal.